File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& a, int k) 
    {
        int zeroes=0,i=0;
        int maxLen=0;
        for(int j=0;j<a.size();j++)
        {
            if(a[j]==0)
            zeroes++;

            if(zeroes>k)
            {
                if(a[i]==0)
                zeroes--;
                i++;
            }
            
            if(zeroes<=k)
            maxLen=max(maxLen,j-i+1);
        }
        return maxLen;
    }
};

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int maxOnes = 0, zeroes = 0, i = 0;
        for(int j = 0; j < nums.length; j++) {
            if(nums[j] == 0){
                zeroes++;
            }
            
            while(zeroes > k){
                if(nums[i] == 0){
                    zeroes--;
                }
                i++;
            }
            maxOnes = Math.max(maxOnes, j - i + 1);
        }
        return maxOnes;
    }
}